Microsoft.Toolkit.Uwp.UI/ImageCache/ImageCache.cs

```diff
@@ -42,10 +42,19 @@ protected class MemoryCacheItem
             public BitmapImage Image { get; set; }
         }
 
+        private class ConcurrentTask
+        {
+            public Task<BitmapImage> Task { get; set; }
+
+            public bool IsAssureOnly { get; set; }
```

Comment:
It indicates the kind of Task. Whether it is only to assure image is in the cache (called by PreCacheAsync, that does not create BitmapImage) or not (called by GetFromCacheAsync). The term "AssurreOnly" is used also as parameter on method GetFromCacheInternalAsync to indicate the kind of task needed. Do you have another suggestion for the name? GetFromCacheInternalAsync should probably be called GetOrAssureAsync?